### PR TITLE
[SPARK-116] query2iRange parallelization

### PR DESCRIPTION
**Pull request by zkhadikova**

### CODE REVIEW

**Comment by srgg** [comment]
This changes should be mentioned in documentation, please introduce this in scope of this PR

**Comment by srgg** [comment]
@zkhadikova the code looks good to me

File: docs/using-connector.md [hunk 1]
```diff
@@ -372,10 +372,27 @@ Any of the Spark Connector options can be provided in `.option()` or `.options()
 ### Key Based Partitioning
 Querying with the following methods with result in a RDD with single partition:
 
-* query2iRange(index, from, to)
 * query2iKeys(index, keys*)
 * queryBucketKeys(keys*)
 
+If 2i is Int, Long or BigInt, range query will be automatically split into a number of subranges defined by **spark.riak.input.split.count** option, creating a partition for each subrange.
```
Comment (korry8911) [inline docs/using-connector.md]: If 2i is an Int, Long or BigInt, the range query will be automatically split into a number of subranges defined by the **spark.riak.input.split.count** option, creating a partition for each subrange.

File: docs/using-connector.md [hunk 2]
```diff
@@ -372,10 +372,27 @@ Any of the Spark Connector options can be provided in `.option()` or `.options()
...
 Querying with the following methods with result in a RDD with single partition:
 
-* query2iRange(index, from, to)
 * query2iKeys(index, keys*)
 * queryBucketKeys(keys*)
 
+If 2i is Int, Long or BigInt, range query will be automatically split into a number of subranges defined by **spark.riak.input.split.count** option, creating a partition for each subrange.  
+* query2iRange(index, from, to)
+
+If 2i is of the other type, single partition will be created.
```
Comment (korry8911) [inline docs/using-connector.md]: If 2i is of the other types, a single partition will be created.

**Pull request merged**